src/test/java/io/jenkins/plugins/analysis/core/quality/HealthReportBuilderTest.java

```diff
@@ -0,0 +1,156 @@
+        HealthReport report = createTestHealthReport(0, 0, Priority.NORMAL, 10, 20, 30);
+
+        assertThat(report).isNull();
+    }
+
+    /**
+     * Tests whether we evaluate correctly to a 100% health if healthy threshold fits the priority issue number.
+     */
+    @Test
+    void testHealthBoundary () {
+        HealthReport reportHighPrio = createTestHealthReport(10, 15, Priority.HIGH,10, 20, 30);
+        HealthReport reportNormalPrio = createTestHealthReport(30, 35, Priority.NORMAL, 10, 20, 30);
+        HealthReport reportLowPrio = createTestHealthReport(60, 65, Priority.LOW, 10, 20, 30);
+
+        assertThat(reportHighPrio.getScore()).isEqualTo(100);
+        assertThat(reportNormalPrio.getScore()).isEqualTo(100);
+        assertThat(reportLowPrio.getScore()).isEqualTo(100);
+    }
+
+    /**
+     * Tests whether we evaluate correctly to a 0% health if unhealthy threshold fits the priority issue number.
+     */
+    @Test
+    void testUnHealthBoundary () {
+        HealthReport reportHighPrio = createTestHealthReport(4, 10, Priority.HIGH, 10, 20, 30);
+        HealthReport reportNormalPrio = createTestHealthReport(15, 30, Priority.NORMAL, 10, 20, 30);
+        HealthReport reportLowPrio = createTestHealthReport(15, 60, Priority.LOW, 10, 20, 30);
+
+        assertThat(reportHighPrio.getScore()).isEqualTo(0);
+        assertThat(reportNormalPrio.getScore()).isEqualTo(0);
+        assertThat(reportLowPrio.getScore()).isEqualTo(0);
+    }
+
+    /**
+     * Tests the correct description for no items.
+     */
+    @Test
+    void testDescriptionNoItem() {
+        HealthReport report = createTestHealthReport(4, 10, Priority.HIGH, 0,0,0);
+        assertThat(report.getDescription()).isEqualTo(DESCRIPTION_MESSAGE + "no" + DESCRIPTION_MESSAGE_ENDING_PLURAL);
+    }
+
+    /**
+     * Tests the correct description for a single item.
+     */
+    @Test
+    void testDescriptionSingleItem() {
+        HealthReport report = createTestHealthReport(4, 10, Priority.HIGH, 1,0,0);
+        assertThat(report.getDescription()).isEqualTo(DESCRIPTION_MESSAGE + "1" + DESCRIPTION_MESSAGE_ENDING_SINGULAR);
+    }
+
+    /**
+     * Tests the correct description for multiple items.
+     */
+    @Test
+    void testDescriptionMultipleItem() {
+        HealthReport report = createTestHealthReport(4, 10, Priority.HIGH, 10,30,60);
+        assertThat(report.getDescription()).isEqualTo(DESCRIPTION_MESSAGE + "10" + DESCRIPTION_MESSAGE_ENDING_PLURAL);
+    }
+
+    /**
+     * Creates the {@link HealthReport} under test with specified parameters.
+     * @param healthyThreshold
+     *      the healthy threshold, i.e. when health is reported as 100%.
+     * @param unhealthyThreshold
+     *      the unhealthy threshold, i.e. when health is reported as 0%.
+     * @param priority
+     *      the minimum priority that should be considered when computing build health
+     * @return the {@link HealthReport} under test
```

Comment:
Wenn JavaDoc, dann alle Parameter beschreiben.

Reply:
done. added missing parameter in JavaDoc